src/ocm/components/clusterConfiguration/staticIp/data/formDataToInfraEnvField.ts

```diff
@@ -0,0 +1,182 @@
+import { dump } from 'js-yaml';
+import { HostStaticNetworkConfig, InfraEnv } from '../../../../../common';
+import {
+  ProtocolVersion,
+  FormViewHost,
+  FormViewNetworkWideValues,
+  StaticFormData,
+  HostIps,
+} from './dataTypes';
+import {
+  FORM_VIEW_PREFIX,
+  getDnsSection,
+  getEthernetInterface,
+  getMachineNetworkFieldName,
+  getNmstateProtocolConfig,
+  getRouteConfig,
+  getVlanInterface,
+  YAML_COMMENT_CHAR,
+} from './nmstateYaml';
+import { getShownProtocolVersions } from './protocolVersion';
+import {
+  Nmstate,
+  NmstateInterface,
+  NmstateProtocolConfigs,
+  NmstateRoutesConfig,
+} from './nmstateTypes';
+import { formDataFromInfraEnvField } from './formDataFromInfraEnvField';
+import { getStaticNetworkConfig } from './fromInfraEnv';
+import { DUMMY_NMSTATE_ADDRESSES, getDummyMacInterfaceMap, getDummyNicName } from './dummyData';
+import { getMachineNetworkCidr } from './machineNetwork';
+import { getEmptyHostIps } from './emptyData';
+
+const REAL_NIC_NAME = 'eth0';
+
+const getPrefixLength = (
+  networkWide: FormViewNetworkWideValues,
+  protocolVerion: ProtocolVersion,
+): number => {
+  const machineNetwork = networkWide.ipConfigs[protocolVerion].machineNetwork;
+  if (machineNetwork.prefixLength === '') {
+    throw 'Machine network prefix length not configured';
+  }
+  return machineNetwork.prefixLength;
+};
+
+const toYamlWithComments = (json: object, comments: string[]) => {
```

Comment:
I'm not sure how reusable this is... Specifically for this feature because there's no backend field I added data to the comments at the beginning of the yaml... I really hope no one will ever use this way to persist data again

Reply:
Then let's keep it simple and we can always extract it later if needed.